BasicClasses/BankAccount.cs

```diff
@@ -128,5 +128,30 @@ public BankAccount(BankAccountType type)
             _ClosedNumber++;
             Type = type;
         }
+
+        public static bool operator ==(BankAccount acc1, BankAccount acc2)
+        {
+            return (acc1.Type == acc2.Type);
```

Comment:
Затлевший в acc1, или acc2 null убьёт вашу программу!
```suggestion
            return Equals(acc1, acc2);
```
Сравнение должно проводиться по номеру счёта, а не по типу!